kong/db/dao/init.lua

```diff
@@ -174,6 +174,62 @@ local function validate_options_value(options, schema, context)
     errors.tags = fmt("cannot be used with '%s'", schema.name)
   end
 
+  if options.pagination ~= nil then
```

Comment:
By the rate this function keeps growing, at one point it will be easier to instead write a schema for `options` and just use the schema library to validate it :)